test/src/module/command/infoTest.c

```diff
@@ -30,6 +30,7 @@ testRun(void)
     {
         StringList *argList = strLstNew();
         strLstAdd(argList, strNewFmt("--repo-path=%s/", strZ(repoPath)));
+        strLstAddZ(argList, "--repo1-retention-full=1");  // avoid warning
```

Comment:
This change on line 33 was not necessary - not sure why it was added and would recommend reverting it.

Reply:
That's because of the `CFGOPT_REPO_RETENTION_FULL_TYPE` in `Data.pm`. I need to know the retention policy in the info command to generate the expected archives list.

Without that line in the tests, we get a warning in the log file.

Maybe there's another way to handle that information without allowing it in the `Data.pm` file ?

Reply:
Yes there is another way. You didn't add any tests in this section (nor the other one) so somewhere in the tests you added, you're missing clearing the logs with harnessLogResult(). So once you figure out where that is, you should remove this additional `strLstAdd(arglist, "--repo1-retention-full=1");`

Reply:
Oh, I indeed didn't thought about clearing the logs between the tests. I'll have a look at that.

I was also wondering if there were any other way to access the retention policy options without changing the `CFGOPT_REPO_RETENTION_FULL_TYPE`,.. in `Data.pm`. Which would avoid the message in the logs..

Reply:
Hi @cmwshang,
Sorry it took me so long to check this.
The `harnessLogResult` is called correctly in the tests I added.

But as I was saying, in info.c, I need to get the `cfgOptRepoRetention*` info. So, I've modified `Data.pm` to make it available for the `info` command. 

The side effect is that if you run the info command without a proper `repo-retention-full` config, you get the warning message in the log. That's why I added the `strLstAdd(arglist, "--repo1-retention-full=1");` to the other tests, not because of not clearing logs.

That's why I was wondering if there were any other way to access the retention policy options without changing the `CFGOPT_REPO_RETENTION_FULL_TYPE`,.. in `Data.pm`.

Kind regards,

Reply:
This warning is raised by the logic here: https://github.com/pgbackrest/pgbackrest/blob/master/src/config/load.c#L215.

So you just need an exception for the info command.

Reply:
Indeed, adding the exception and removing the unneeded test line seems to work as expected (commit 
24a1dd8).

Thanks for the tip!